core/elf.cc

```diff
@@ -1553,6 +1551,14 @@ program::get_library(std::string name, std::vector<std::string> extra_path, bool
     auto ret = load_object(name, extra_path, loaded_objects, dlopen);
     _loaded_objects_stack.push(loaded_objects);
 
+    // Relocate objects *after* loading all objects, otherwise, we cannot resolve circluar relocations.
+    for (auto ef : loaded_objects) {
+        if (!ef->is_statically_linked_executable() && !ef->is_linux_dl()) {
+            ef->relocate();
+            ef->fix_permissions();
+        }
+    }
+
```

Comment:
Looks good, thanks! Please move the comment "Do not relocate static executables as they are linked with its own dynamic linker." to this new place as it explains the if(). The original comment probably needs to be edited, as half of it (the half about relocation) is no longer relevant in its original position.

Reply:
Good catch, updated the comments accordingly.